packages/amesos2/src/Amesos2_MatrixAdapter_decl.hpp

```diff
@@ -131,35 +136,47 @@ namespace Amesos2 {
      * \exception std::runtime_error Thrown if there is an error while extracting
      * row values from the underlying matrix.
      */
+    #ifdef TPETRA_ENABLE_DEPRECATED_CODE
```

Comment:
@iyamazaki thanks for the work on this PR, I have a couple questions on some Amesos2 changes:

1. There are several locations where routines like this `getCrs` are guarded based on enabled deprecated code but there is not an else branch, why does these need to be removed? I'm not involved with the Tpetra updates, it's not clear to me why/if this is necessary to coincide with those changes or just considered clean up
2. Can you confirm which solvers were enabled in testing, and was this PR tested with/without MPI? Some of the remnant non-Kokkos view routines may be necessary to support matrix inputs with non-contiguous GIDs, which are only tested with MPI enabled. If you could confirm the following KLU2, Basker and SuperLU tests ran and passed with these changes, that would help me feel comfortable with the removals (I think only KLU2 is enabled in PR testing):
https://github.com/trilinos/Trilinos/blob/a6d3274e2f75876e7d682e34f92b12dded96f014/packages/amesos2/test/solvers/Basker_UnitTests.cpp#L389
https://github.com/trilinos/Trilinos/blob/a6d3274e2f75876e7d682e34f92b12dded96f014/packages/amesos2/test/solvers/KLU2_UnitTests.cpp#L419
https://github.com/trilinos/Trilinos/blob/a6d3274e2f75876e7d682e34f92b12dded96f014/packages/amesos2/test/solvers/Superlu_UnitTests.cpp#L374
https://github.com/trilinos/Trilinos/blob/a6d3274e2f75876e7d682e34f92b12dded96f014/packages/amesos2/test/solvers/ShyLUBasker_UnitTests.cpp#L425

Reply:
This is an *excellent* point.  The path I took in Ifpack2 is to use the deprecated Tpetra interfaces *only* when I was inheriting from a Tpetra object which had deprecated interfaces.  Since Amesos2 is not doing this, those interfaces can probably go.

Reply:
Thank you for the review, @ndellingwood!! The most of the changes for Amesos2 are because Tpetra is switching to return the local Crs as Kokkos::views instead of Teuchos::ArrayViews. There are public slides that summarize the MV changes, but I am not sure if there are slides about Crs changes. I can ask @kddevin, if they would help.

I have been testing with KLU2 and SuperLU, but not with Baskers, using MPI. I updated the motivation above, and the [Issue#8761](https://github.com/trilinos/Trilinos/issues/8761) with my Amesos2 configure.

Reply:
Thank you for the inputs, @csiefer2. For now, we kept the old interfaces, which use the deprecated Tpetra interfaces, as deprecated codes because I was not sure if any solvers or other codes use these old adapter interfaces. So, if they do, these solvers need to enable the Tpetra's deprecated code. But, I can just remove them, if that is better!